etcdserver/apply.go

```diff
@@ -59,7 +59,7 @@ type applierV3 interface {
 
 	AuthEnable() (*pb.AuthEnableResponse, error)
 	AuthDisable() (*pb.AuthDisableResponse, error)
-	Authenticate(r *pb.AuthenticateRequest) (*pb.AuthenticateResponse, error)
+	Authenticate(index uint64, r *pb.InternalAuthenticateRequest) (*pb.AuthenticateResponse, error)
```

Comment:
i think the interface should be generic enough. I also plan to decouple the pb request conversion. so the auth pkg does not need to understand etcd request anymore.